TestingPriceCalculater/PriceCalculaterTesting.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.IO;
+using Xunit;
+namespace TestingPriceCalculater
+{
+
+    public class TaxTest
+    {   Product product;
+        Calculater calculater1;
+        ITaxService MyTax;
+        public TaxTest()
+        {
+            product = new Product
+            {
+                Name = "mybook",
+                UPC = 123,
+                Price = 20.25M
+            };
+            calculater1 = new Calculater();
+        }
+        [Theory]
+        [InlineData(20,24.30)]
+        [InlineData(21, 24.50)]
+        public void TestFinalPrice1(int percentage ,decimal expected )
+        {
+            MyTax = new TaxService(percentage);
```

Comment:
I prefer to have only one main class (like PriceCaclulator) that the consumer use to interact with the calculator, and all interactions and dependency are implemented in that main class